code/datums/components/slippery.dm

```diff
@@ -1,42 +1,155 @@
 	SIGNAL_HANDLER
-
 	if(!isliving(arrived))
 		return
 	var/mob/living/victim = arrived
-	if(!(victim.movement_type & FLYING) && victim.slip(knockdown_time, parent, lube_flags, paralyze_time, force_drop_items) && callback)
+	if(!(victim.movement_type & (FLYING | FLOATING)) && victim.slip(knockdown_time, parent, lube_flags, paralyze_time, force_drop_items) && callback)
 		callback.Invoke(victim)
 
+/*
+ * Gets called when COMSIG_ITEM_EQUIPPED is sent to parent.
+ * This proc register slip signals to the equipper.
+ * If we have a slot whitelist, we only register the signals if the slot is valid (ex: clown PDA only slips in ID or belt slot).
+ *
+ * source - the source of the signal
+ * equipper - the mob we're equipping the slippery thing to
+ * slot - the slot we're equipping the slippery thing to on the equipper.
+ */
+/datum/component/slippery/proc/on_equip(datum/source, mob/equipper, slot)
+	SIGNAL_HANDLER
+
+	if((!LAZYLEN(slot_whitelist) || (slot in slot_whitelist)) && isliving(equipper))
+		holder = equipper
+		qdel(GetComponent(/datum/component/connect_loc_behalf))
+		AddComponent(/datum/component/connect_loc_behalf, holder, holder_connections)
+		RegisterSignal(holder, COMSIG_PARENT_QDELETING, .proc/holder_deleted)
+
+/*
+ * Detects if the holder mob is deleted.
+ * If our holder mob is the holder set in this component, we null it.
+ *
+ * source - the source of the signal
+ * possible_holder - the mob being deleted.
+ */
+/datum/component/slippery/proc/holder_deleted(datum/source, datum/possible_holder)
+	SIGNAL_HANDLER
+
+	if(possible_holder == holder)
+		holder = null
+
+/*
+ * Gets called when COMSIG_ITEM_DROPPED is sent to parent.
+ * Makes our holder mob un-slippery.
+ *
+ * source - the source of the signal
+ * user - the mob that was formerly wearing our slippery item.
+ */
+/datum/component/slippery/proc/on_drop(datum/source, mob/user)
+	SIGNAL_HANDLER
+
+	UnregisterSignal(user, COMSIG_PARENT_QDELETING)
+
+	qdel(GetComponent(/datum/component/connect_loc_behalf))
+	add_connect_loc_behalf_to_parent()
+
+	holder = null
+
+/*
+ * The slip proc, but for equipped items.
+ * Slips the person who crossed us if we're lying down and unbuckled.
+ *
+ * source - the source of the signal
+ * AM - the atom/movable that slipped on us.
+ */
+/datum/component/slippery/proc/Slip_on_wearer(datum/source, atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	SIGNAL_HANDLER
+	if(!(holder.mobility_flags & MOBILITY_STAND) && !holder.buckled)
+		Slip(source, arrived)
+
 /datum/component/slippery/UnregisterFromParent()
 	. = ..()
 	qdel(GetComponent(/datum/component/connect_loc_behalf))
+
+/// Used for making the clown PDA only slip if the clown is wearing his shoes and the elusive banana-skin belt
+/datum/component/slippery/clowning
+
+/datum/component/slippery/clowning/Slip_on_wearer(datum/source, atom/movable/AM)
+	var/obj/item/I = holder.get_item_by_slot(ITEM_SLOT_FEET)
```

Comment:
Confirm there is an item in that slot or else there is a chance you'll be checking nulls.